### PR TITLE
Add caveat for opApply and exceptions.

### PR DESCRIPTION
**Pull request by Shachar**
As discussed on the forum, add explanation about exception handling inside opApply, and their interaction with exceptions thrown from within the opApply delelgate.

### CODE REVIEW

**Comment by schveiguy** [comment]
Link to discussion: https://forum.dlang.org/post/n8ptpu$1ncu$1@digitalmars.com

File: spec/statement.dd [hunk 1]
```diff
@@ -682,6 +682,12 @@ $(CONSOLE
         }
         --------------
 
+    $(P It is important to make sure that, if $(D opApply) catches any exceptions, that those
+        exceptions did not origin at the delegate passed to $(I opApply). Catching an exception
+        that escaped from the delegate would cause a condition that the user expected would terminate
+        the loop to not do so.
```
Comment (schveiguy) [inline spec/statement.dd]: 1. "origin at" -> originate from
2. The last sentence is cryptic. Possibly better: "The user would expect exceptions thrown from a `foreach` body to both terminate the loop, and propagate outside the `foreach` body."

Comment (schuetzm) [inline spec/statement.dd]: Also, _that_ appears twice: "make sure **that**, ..., **that** those"

**Comment by adamdruppe** [comment]
@Shachar  if you update the wording a bit from the comments, i'll pull this.

**Comment by AndrejMitrovic** [comment]
Hi @Shachar, could you amend the suggested fixes? If you do not have the time we can take over. Thanks!

**Comment by andralex** [comment]
I'll pull now and make the changes as a fixup.

**Comment by andralex** [comment]
https://github.com/dlang/dlang.org/commit/ca83d45b1d8e7576639faac50f67d5bce0737277

**Pull request merged**